mcstatus/status_response.py

```diff
@@ -0,0 +1,419 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, List, Optional, TYPE_CHECKING, Tuple, Union
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+from mcstatus.utils import deprecated
+
+__all__ = [
+    "STYLE_MAP",
+    "AbstractDataclass",
+    "MCStatusResponse",
+    "JavaStatusResponse",
```

Comment:
I exposed this things because `STYLE_MAP` can be useful for user, and `AbstractDataClass` can be used in another parts of code. Of course, it's not very suitable here, but I don't actually know where I should place it.

Reply:
People can still technically import STYLE_MAP, even if it isn't listed in `__all__`, but it isn't the job of mcstatus to provide people with things like these and it shouldn't be publically exposed, because we may change this in the future a lot, and I really don't like adding the burden of making it backwards compatible. Mcstatus is here purely to make it easy to get the response from status (or query) protocols, not to give users objects like these to help them resolve colors from an arbitrary response. That's something that could make sense in a project like pymine-net, which tries to completely implement everything in minecraft's protocol.

Reply:
And what about `AbstractDataclass`?
Removed `STYLE_MAP` from `__all__` in ee7ef8cb78d9983da3da7b7e6bedcda6cbe87ef9.

Reply:
I don't think there's a need to expose that either. It's fully internal to this file and it can still be used from other mcstatus files without being listed in `__all__`. This should be preferred because exposing it as part of public API of mcstatus makes no sense, this library isn't here for providing this class, it's not even remotely related to the goal of this project and so it certainly shouldn't be a part of public API

Reply:
> it can still be used from other mcstatus files without being listed in __all__

`pyright` can triger on not listed imports.

Reply:
nevertheless, this class probably won't be used anywhere else anyway, and this can be an issue once we would want to use it, for now it shouldn't be incuded

Reply:
Deleted in 1b59b95d6ab18c85b3f13bb87ade307895b34283.